ui-modules/blueprint-composer/app/components/providers/blueprint-service.provider.js

```diff
@@ -520,6 +523,70 @@ function BlueprintService($log, $q, $sce, paletteApi, iconGenerator, dslService,
+                    case '$brooklyn:object':
+                        if(args["type"] != null && EXTRA_CONSTRAINTS.includes(args["factoryMethod.name"])) {
+                            let fct = args["factoryMethod.name"];
+                            let constrVals = args["factoryMethod.args"];
+                            switch (fct) {
+                                case 'equal':
+                                    if (!isSet() || isSet() && val() !== constrVals[0]) {
+                                        message = `<samp>${config.name}</samp> must be equal to ${constrVals[0]}`;
+                                    }
+                                    break;
+                                case 'greater_than':
+                                    if (!isSet() || isSet() && val() <= constrVals[0]) {
+                                        message = `<samp>${config.name}</samp> must be greater than ${constrVals[0]}`;
+                                    }
+                                    break;
+                                case 'greater_than_or_equal':
+                                    if (!isSet() || isSet() && val() < constrVals[0]) {
+                                        message = `<samp>${config.name}</samp> must be greater than or equal to ${constrVals[0]}`;
+                                    }
+                                    break;
+                                case 'less_than':
+                                    if (!isSet() || isSet() && val() >= constrVals[0]) {
+                                        message = `<samp>${config.name}</samp> must be less than ${constrVals[0]}`;
+                                    }
+                                    break;
+                                case 'less_than_or_equal':
+                                    if (!isSet() || isSet() && val() > constrVals[0]) {
+                                        message = `<samp>${config.name}</samp> must be less than or equal to ${constrVals[0]}`;
+                                    }
+                                    break;
+                                case 'in_range':
+                                    if (!isSet() || isSet() && val() >= constrVals[0] && val() <= constrVals[1]) {
+                                        message = `<samp>${config.name}</samp> must be in range of (inclusive) [${constrVals[0]} , ${constrVals[1]}]`;
+                                    }
+                                    break;
+                                case 'valid_values':
+                                    if (!isSet() || isSet() && constrVals[0].includes(val())) {
+                                        message = `<samp>${config.name}</samp> must be one of (${constrVals.toString()})`;
+                                    }
+                                    break;
+                                case 'length':
+                                    if (!isSet() || isSet() && val().length === constrVals[0]) {
```

Comment:
think this should be `val.length() !===`

Reply:
Yup, good catch! Thx!